src/app.tsx

```diff
@@ -0,0 +1,105 @@
+// tslint:disable-next-line:no-implicit-dependencies
+import { Opportunity } from "@src/generated/sobs";
+import { Affix, Button, Col, Icon, Layout, Row, Spin } from "antd";
+import * as React from "react";
+
+interface Props {
+    id: string;
+}
+
+interface State {
+    isOpptyLoading: boolean;
+    isTouchSaving: boolean;
+    oppName: string;
+    touchCount: number;
+}
+
+const loadingIndicator = (
+    <Affix offsetTop={300}>
+        <Row type="flex" justify="center">
+            <Col span={1}>
+                <Icon spin={true} style={{ fontSize: 56 }} type="loading" />
+            </Col>
+        </Row>
+    </Affix>
+);
+
+export class App extends React.Component<Props, State> {
+    constructor(props) {
+        super(props);
+        this.state = {
+            oppName: "Loading",
+            isOpptyLoading: true,
+            isTouchSaving: false,
+            touchCount: 0,
+        };
+        this.touchOppty = this.touchOppty.bind(this);
```

Comment:
@ChuckJonas so i'm a bit confused, I thought that wasn't advisable since it would lead to the event handler function being re-initialized everytime the component was rendered ... 

came to it with this link https://stackoverflow.com/questions/36677733/why-shouldnt-jsx-props-use-arrow-functions-or-bind in your tslint.json file

does that not apply if I do thinks like you're describing?